core/types/rollup_cost.go

```diff
@@ -179,6 +190,25 @@ func NewL1CostFunc(config *params.ChainConfig, statedb StateGetter) L1CostFunc {
 	}
 }
 
+// NewOperatorCostFunc returns a function used for calculating operator fees, or nil if this is
+// not an op-stack chain.
+func NewOperatorCostFunc(config *params.ChainConfig, statedb StateGetter) OperatorCostFunc {
+	if config.Optimism == nil {
+		return nil
+	}
+	return func(gas *big.Int, blockTime uint64) *big.Int {
+		if !config.IsOptimismIsthmus(blockTime) {
+			return big.NewInt(0)
+		}
+		operatorFeeParams := statedb.GetState(L1BlockAddr, OperatorFeeParamsSlot).Bytes()
+
+		operatorFeeScalar, operatorFeeConstant := extractOperatorFeeParams(operatorFeeParams)
+		product := operatorFeeScalar.Mul(gas, operatorFeeScalar)
+		product = product.Div(product, oneMillion)
+		return product.Add(product, operatorFeeConstant)
```

Comment:
This is the place where i am most concerned about an overflow. We should cap it at max u256. Then downstream i dont think its possible to have an overflow

Reply:
this should be added to the specs that the value cannot go above u256. we should also consider checking for overflows in the gas price oracle contract. its a bit of a edge case but we dont want to allow the chain operator to be able to trigger a chain split

Reply:
We have established that the highest operator fee is an int with 77 bits, so an overflow is not possible. Does that cover this discussion?